modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPOptionLocalServiceImpl.java

```diff
@@ -149,6 +154,22 @@ public CPOption deleteCPOption(CPOption cpOption) throws PortalException {
 		_cpOptionValueLocalService.deleteCPOptionValues(
 			cpOption.getCPOptionId());
 
+		// Commerce product option rels
+
+		CPDefinitionOptionRelLocalService cpDefinitionOptionRelLocalService =
+			_cpDefinitionOptionRelLocalServiceSnapshot.get();
+
+		List<CPDefinitionOptionRel> cpDefinitionOptionRels =
+			cpDefinitionOptionRelLocalService.getCPOptionCPDefinitionOptionRels(
+				cpOption.getCPOptionId());
+
+		for (CPDefinitionOptionRel cpDefinitionOptionRel :
```

Comment:
I've just talked with Giulia, we think that the best is to avoid the deletion of the option template if the option is used in some products. So add the count method of this query in place of this and make the check before the deletion, so if count > 0 don't delete and throw an exception, this exception should be shown on screen with a session error